File: server/pkg/model/webhook.go
package model

type ConvWebhook struct {
	ID            int64           `json:"id"`
	ConvID        string          `json:"conv_id"`
	Name          string          `json:"name"`
	APIKeyPlain   string          `json:"api_key,omitempty"`
	APIKeyHash    string          `json:"-"`
	CallbackURL   string          `json:"callback_url,omitempty"`
	Headers       []WebhookHeader `json:"headers,omitempty"`
	CIDRWhitelist []string        `json:"cidr_whitelist,omitempty"`
	CreatedBy     string          `json:"created_by"`
	CreatedAt     int64           `json:"created_at"`
}

type WebhookHeader struct {
	Key   string `json:"key"`
	Value string `json:"value"`
}

File: server/web/src/types/webhook.ts
export interface ConvWebhook {
  id: number
  conv_id: string
  name: string
  api_key?: string
  callback_url?: string
  headers?: { key: string; value: string }[]
  cidr_whitelist?: string[]
  created_by: string
  created_at: number
}
